### DISCUSSION TITLE
RequestExecutor Global Application ID?

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#15083] [Q&A] [answered]
In the JVM version of the client source code, [line 58](https://github.com/ravendb/ravendb-jvm-client/blob/v5.4/src/main/java/net/ravendb/client/http/RequestExecutor.java#L58)  (and I assume others) there is a GLOBAL_APPLICATION_ID at the top of the file that's initialized to a random UUID. What is this used for? I see it being passed into the UpdateTopologyParameters ([line 298](https://github.com/ravendb/ravendb-jvm-client/blob/v5.4/src/main/java/net/ravendb/client/http/RequestExecutor.java#L298)/[line 576](https://github.com/ravendb/ravendb-jvm-client/blob/v5.4/src/main/java/net/ravendb/client/http/RequestExecutor.java#L576)) and therefore into updateTopologyAsync(). However, it's not clear what the purpose behind this ID is for.

Is this ID settable by the user somewhere during the construction of the document store or via an environment variable or some other mechanism? Is it communicated to the ravendb server?
Thanks.

**Comment by ayende** [comment, marked as answer]
This is meant to prevent a common mistake.
You are expected to create the `DocumentStore` once and use if for a long time. Usually as a singleton.
You are **not** expected to create it once per request.

A common API issue is something like this:


```
public User GetUserById(string id)
{
      using var store = new DocumentStore{ Urls = ... , Certificate = ... };
      using var session = store.OpenSession();
      return session.Load<User>(id);
}
```

What this does is to create a new document store per operation. But that is a bad thing (caching, connection pooling, etc are all at the level of the document store).

We use the global application id to detect that and show the user an alert.
This is done by detecting the number of new topology requests from the same process to the same database on the server side.

You are expected to just generate a new id per process.

**Reply by damccull** [reply]
Great! That's a cool idea for using a uuid.
